src/wayland/shortcuts_inhibit/inhibitor.cpp

```diff
@@ -147,19 +128,15 @@ void ShortcutsInhibitor::onWaylandSurfaceCreated() {
 	}
 
 	if (this->inhibitor) {
-		QObject::disconnect(this->inhibitor, nullptr, this, nullptr);
-		manager->unrefShortcutsInhibitor(this->inhibitor);
-		this->inhibitor = nullptr;
+		qFatal("ShortcutsInhibitor: inhibitor already exists when creating surface");
 	}
+
 	this->inhibitor = manager->createShortcutsInhibitor(this->mWaylandWindow);
 
 	if (this->inhibitor) {
-		QObject::connect(
-		    this->inhibitor,
-		    &impl::ShortcutsInhibitor::activeChanged,
-		    this,
-		    &ShortcutsInhibitor::activeChanged
-		);
+		this->bActive.setBinding([this]() {
+			return this->inhibitor ? this->inhibitor->bindableActive().value() : false;
+		});
```

Comment:
inhibitor must be a bindable for this to work, and there shouldnt be a reason to set this anywhere but the constructor.